### PR TITLE
[SPARK-35095][SS][TESTS] Use ANSI intervals in streaming join tests

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR extends the following tests to use day-time intervals.

* StreamingOuterJoinSuite.right outer with watermark range condition
* StreamingOuterJoinSuite.left outer with watermark range condition

### Why are the changes needed?

Currently, there are no tests to use day-time intervals.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New assertions.

### CODE REVIEW

**Comment by sarutak** [comment]
cc: @MaxGekk

**Comment by sarutak** [comment]
retest this please.

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala [hunk 1]
```diff
@@ -777,6 +781,35 @@ class StreamingOuterJoinSuite extends StreamingJoinSuite {
         CheckNewAnswer(outerResult),
         assertNumStateRows(total = 2, updated = 1)
       )
+
+      Seq(
+        ("10 minutes",
+          "interval 3 minutes 30 seconds"),
+        ("10 minutes",
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala:788]: We should support watermarks as ANSI intervals in the future. @sarutak Could you open an JIRA for that, so, we won't forget to implement it.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @sarutak .

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala (hunk 1 above)
Comment (sarutak) [reply sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala:788]: O.K, I'll open it.

**Pull request closed without merging**